lib/internal/loader/ModuleRequest.js

```diff
@@ -95,14 +95,27 @@ exports.resolve = (specifier, parentURL) => {
                            url.protocol, 'file:');
   }
 
-  if (!preserveSymlinks) {
-    const real = realpathSync(internalURLModule.getPathFromURL(url), {
-      [internalFS.realpathCacheKey]: realpathCache
-    });
-    const old = url;
-    url = internalURLModule.getURLFromFilePath(real);
-    url.search = old.search;
-    url.hash = old.hash;
+  try {
+    if (!preserveSymlinks) {
+      let real;
+      real = realpathSync(internalURLModule.getPathFromURL(url), {
```

Comment:
Why did this change to `let real` instead of staying `const real`? It's still only assigned once during its lifetime.